test/7.compliance.test.js

```diff
@@ -1,124 +1,140 @@
 import {expect} from 'chai';
-import FL from 'fantasy-land';
 import Future from '../index.mjs.js';
-import * as U from './util';
-import jsc from 'jsverify';
+import {assertEqual as eq, I, B, noop, STACKSIZE, error, bang} from './util';
+import {Functor, Bifunctor, Apply, Applicative, Chain, ChainRec, Monad} from 'fantasy-laws';
+import Z from 'sanctuary-type-classes';
+import {
+  bool,
+  constant as _k,
+  falsy,
+  fn,
+  letrec,
+  nat,
+  number,
+  oneof,
+  property,
+  string
+} from 'jsverify';
+
+//  FutureArb :: Arbitrary a -> Arbitrary b -> Arbitrary (Future a b)
+var FutureArb = function(larb, rarb){
+  var toStr = function(m){ return m.toString() };
+  var getLeft = function(m){
+    var x;
+    m.fork(function(y){ x = y }, function(){ throw error });
+    return x;
+  };
+  var getRight = function(m){
+    var x;
+    m.fork(function(){ throw error }, function(y){ x = y });
+    return x;
+  };
+  return oneof(
+    larb.smap(Future.reject, getLeft, toStr),
+    rarb.smap(Future.of, getRight, toStr),
+    larb.smap(function(x){ return Future.reject(x).map(I) }, getLeft, toStr),
+    rarb.smap(function(x){ return Future.of(x).map(I) }, getLeft, toStr)
```

Comment:
The reason for the latter two Arbitraries in this union is to greatly increase the code coverage made by these tests, because *just* `reject` or `of`-based Futures have optimised implementations of many of the algebras. Mapping over them once with `I` removes this layer of optimisation and falls back to the default implementations.